python-package/setup.py

```diff
@@ -28,6 +28,7 @@
       install_requires=[
           'numpy',
           'scipy',
+          'scikit-learn',
```

Comment:
It should be on suggests. As sklearn module will work when sklearn is available. But XGBoost will function without sklearn